src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -96,13 +137,415 @@ void AreaTrigger::Update(uint32 p_time)
         Remove(); // expired
 
     WorldObject::Update(p_time);
+    _timeSinceCreated += p_time;
```

Comment:
Should check for negative duration, and skip checks for it @line 134 (it will instantly remove areatriggers with duration = -1)